processing/src/main/java/org/apache/druid/query/aggregation/any/StringAnyBufferAggregator.java

```diff
@@ -65,6 +67,27 @@ public void aggregate(ByteBuffer buf, int position)
     }
   }
 
+  private String readValue(Object object)
+  {
+    if (object == null) {
+      return null;
+    }
+    if (object instanceof List) {
+      List<Object> objectList = (List) object;
+      if (objectList.size() == 0) {
+        return null;
+      }
+      if (objectList.size() == 1) {
+        return DimensionHandlerUtils.convertObjectToString(objectList.get(0));
+      }
+      if (aggregateMultipleValues) {
+        return DimensionHandlerUtils.convertObjectToString(objectList);
+      }
+      return DimensionHandlerUtils.convertObjectToString(objectList.get(0));
```

Comment:
@clintropolis Thank you Clint. Is it ok to leave this extra safety check in the code ?

Reply:
if this was ... any (heh!) other aggregator i would say change it (and probably re-arrange the if) because aggregate is in a hot path, but since this one basically only calls aggregate once and then short-circuits the rest of the time... so its probably fine